packages/@angular/cli/tasks/eject.ts

```diff
@@ -0,0 +1,473 @@
+    config['output'] = this._outputReplacer(config['output']);
+    config['module'] = this._moduleReplacer(config['module']);
+    config['context'] = undefined;
+
+    return JSON.stringify(config, (k, v) => this._replacer(k, v), 2)
+      .replace(/"\uFF01(.*?)\uFF01"/g, (_, v) => {
+        return JSON.parse(`"${v}"`);
+      })
+      .replace(/(\s*)(.*?)"\uFF02(.*?)\uFF02"(,?).*/g, (_, indent, key, value, comma) => {
+        const ctor = JSON.parse(`"${value}"`).split(/\n+/g).join(indent);
+        return `${indent}${key}new ${ctor}${comma}`;
+      })
+      .replace(/"\uFF01(.*?)\uFF01"/g, (_, v) => {
+        return JSON.parse(`"${v}"`);
+      });
+  }
+
+  generateVariables(): string {
+    let variableOutput = '';
+    Object.keys(this.variableImports)
+      .forEach((key: string) => {
+        const [module, name] = key.split(/\./);
+        variableOutput += `const ${this.variableImports[key]} = require` + `('${module}')`;
+        if (name) {
+          variableOutput += '.' + name;
+        }
+        variableOutput += ';\n';
+      });
+    variableOutput += '\n';
+
+    Object.keys(this.imports)
+      .forEach((key: string) => {
+        const [module, name] = key.split(/\./);
+        variableOutput += `const { ${this.imports[key].join(', ')} } = require` + `('${module}')`;
+        if (name) {
+          variableOutput += '.' + name;
+        }
+        variableOutput += ';\n';
+      });
+    variableOutput += '\n';
+
+    Object.keys(this.variables)
+      .forEach((key: string) => {
+        variableOutput += `const ${key} = ${this.variables[key]};\n`;
+      });
+    variableOutput += '\n\n';
+
+    return variableOutput;
+  }
+}
+
+
+export default Task.extend({
+  run: function (runTaskOptions: EjectTaskOptions) {
+    const project = this.cliProject;
+    const cliConfig = CliConfig.fromProject();
+    const config = cliConfig.config;
+    const outputPath = runTaskOptions.outputPath || config.apps[0].outDir;
+    const force = runTaskOptions.force;
+
+    if (project.root === outputPath) {
+      throw new SilentError ('Output path MUST not be project root directory!');
+    }
+
+    const webpackConfig = new NgCliWebpackConfig(runTaskOptions).config;
+    const serializer = new JsonWebpackSerializer(process.cwd(), outputPath);
+    const output = serializer.serialize(webpackConfig);
+    const webpackConfigStr = `${serializer.generateVariables()}\n\nmodule.exports = ${output};\n`;
+
+    return Promise.resolve()
+      .then(() => exists('webpack.config.js'))
+      .then(webpackConfigExists => {
+        if (webpackConfigExists && !force) {
+          throw new SilentError('The webpack.config.js file already exists.');
+        }
+      })
+      // Read the package.json and update it to include npm scripts. We do this first so that if
+      // an error already exists
+      .then(() => ts.sys.readFile('package.json'))
+      .then(packageJson => JSON.parse(packageJson))
+      .then(packageJson => {
```

Comment:
Consider making `packageJson` of type `any` which will allow for direct property access instead of strings in case someone creates a type later on.